src/NonceTracker.ts

```diff
@@ -1,5 +1,6 @@
 import assert from 'assert';
 import { Mutex } from 'async-mutex';
+import type { SafeEventEmitterProvider } from '@metamask/eth-json-rpc-provider';
```

Comment:
Is there an easy way to do that?

Reply:
Ah sorry I didn't see this comment before.

Because `@metamask/eth-json-rpc-provider` shows up in the type declaration, it can't be in `devDependencies`, it has to be in `dependencies` instead. This ensures that TypeScript consumers automatically get the type definitions for this import and do not have to add it themselves.

I know that we've discussed splitting this type off before. I am not opposed to that, I am just unsure where. Also, I suspect that most of these transitive dependencies are coming from `@metamask/utils`. I will open a discussion in chat for this since I feel like it might be easier to talk there.